state/peerAccount.go

```diff
@@ -108,7 +108,11 @@ func (pa *peerAccount) SetTempRating(rating uint32) {
 }
 
 // SetListAndIndex will update the peer's list (eligible, waiting) and the index inside it with journal
-func (pa *peerAccount) SetListAndIndex(shardID uint32, list string, index uint32) {
+func (pa *peerAccount) SetListAndIndex(shardID uint32, list string, index uint32, updatePreviousList bool) {
+	if updatePreviousList {
```

Comment:
I think it is better to have a peerAccountV2 - where this update is true - and leave the old code as it is.

Reply:
And you do not need to push the boolean in the argument list.

Reply:
I would leave the existing structure with extension field instead of creating a new struct. This is also related to my comment above.